File: src/components/JSONHandler/src/ALRPCObjectsImpl/ButtonCapabilities.cpp
#include "../../include/JSONHandler/ALRPCObjects/ButtonCapabilities.h"
#include "ButtonCapabilitiesMarshaller.h"
#include "ButtonNameMarshaller.h"
/*
  interface	Ford Sync RAPI
  version	1.2
  date		2011-05-17
  generated at	Tue Oct 30 08:29:32 2012
  source stamp	Thu Oct 25 06:49:27 2012
  author	robok0der
*/



ButtonCapabilities::ButtonCapabilities(const ButtonCapabilities& c)
{
  *this=c;
}


bool ButtonCapabilities::checkIntegrity(void)
{
  return ButtonCapabilitiesMarshaller::checkIntegrity(*this);
}


ButtonCapabilities::ButtonCapabilities(void)
{
}



bool ButtonCapabilities::set_longPressAvailable(bool longPressAvailable_)
{
  longPressAvailable=longPressAvailable_;
  return true;
}

bool ButtonCapabilities::set_name(const ButtonName& name_)
{
  if(!ButtonNameMarshaller::checkIntegrityConst(name_))   return false;
  name=name_;
  return true;
}

bool ButtonCapabilities::set_shortPressAvailable(bool shortPressAvailable_)
{
  shortPressAvailable=shortPressAvailable_;
  return true;
}

bool ButtonCapabilities::set_upDownAvailable(bool upDownAvailable_)
{
  upDownAvailable=upDownAvailable_;
  return true;
}




bool ButtonCapabilities::get_longPressAvailable(void) const
{
  return longPressAvailable;
}


const ButtonName& ButtonCapabilities::get_name(void) const 
{
  return name;
}


bool ButtonCapabilities::get_shortPressAvailable(void) const
{
  return shortPressAvailable;
}


bool ButtonCapabilities::get_upDownAvailable(void) const
{
  return upDownAvailable;
}

File: mobile/android/SyncProxyAndroid/src/com/ford/syncV4/proxy/rpc/ButtonCapabilities.java
package com.ford.syncV4.proxy.rpc;

import java.util.Hashtable;

import com.ford.syncV4.proxy.RPCStruct;
import com.ford.syncV4.proxy.constants.Names;
import com.ford.syncV4.proxy.rpc.enums.ButtonName;
import com.ford.syncV4.util.DebugTool;

public class ButtonCapabilities extends RPCStruct {

    public ButtonCapabilities() { }
    public ButtonCapabilities(Hashtable hash) {
        super(hash);
    }
    public ButtonName getName() {
        Object obj = store.get(Names.name);
        if (obj instanceof ButtonName) {
            return (ButtonName) obj;
        } else if (obj instanceof String) {
            ButtonName theCode = null;
            try {
                theCode = ButtonName.valueForString((String) obj);
            } catch (Exception e) {
                DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.name, e);
            }
            return theCode;
        }
        return null;
    }
    public void setName( ButtonName name ) {
        if (name != null) {
            store.put(Names.name, name );
        }
    }
    public Boolean getShortPressAvailable() {
        return (Boolean) store.get( Names.shortPressAvailable );
    }
    public void setShortPressAvailable( Boolean shortPressAvailable ) {
        if (shortPressAvailable != null) {
            store.put(Names.shortPressAvailable, shortPressAvailable );
        }
    }
    public Boolean getLongPressAvailable() {
        return (Boolean) store.get( Names.longPressAvailable );
    }
    public void setLongPressAvailable( Boolean longPressAvailable ) {
        if (longPressAvailable != null) {
            store.put(Names.longPressAvailable, longPressAvailable );
        }
    }
    public Boolean getUpDownAvailable() {
        return (Boolean) store.get( Names.upDownAvailable );
    }
    public void setUpDownAvailable( Boolean upDownAvailable ) {
        if (upDownAvailable != null) {
            store.put(Names.upDownAvailable, upDownAvailable );
        }
    }
}
